File: CS/MySolution.Module.Win/ButtonDetailViewItemWin.cs
using System;
using System.Windows.Forms;
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp;

namespace MySolution.Module.Win {
    public interface IModelButtonDetailViewItemWin : IModelViewItem { }

    [ViewItemAttribute(typeof(IModelButtonDetailViewItemWin))]
    public class ButtonDetailViewItemWin : ViewItem {
        public ButtonDetailViewItemWin(IModelViewItem model, Type objectType)
            : base(objectType, model.Id) {
            CreateControl();
        }
        protected override object CreateControlCore() {
            Button button = new Button();
            button.Text = "Click me!";
            button.Click += new EventHandler(button_Click);
            return button;
        }
        void button_Click(object sender, EventArgs e) {
            throw new UserFriendlyException("Action from custom view item was executed!");
        }
    }
}

File: VB/MySolution.Module.Win/ButtonDetailViewItemWin.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp

Namespace MySolution.Module.Win

    Public Interface IModelButtonDetailViewItemWin
        Inherits IModelViewItem

    End Interface

    <ViewItemAttribute(GetType(IModelButtonDetailViewItemWin))>
    Public Class ButtonDetailViewItemWin
        Inherits ViewItem

        Public Sub New(ByVal model As IModelViewItem, ByVal objectType As Type)
            MyBase.New(objectType, model.Id)
        End Sub

        Protected Overrides Function CreateControlCore() As Object
            Dim button As Button = New Button()
            button.Text = "Click me!"
            AddHandler button.Click, New EventHandler(AddressOf button_Click)
            Return button
        End Function

        Private Sub button_Click(ByVal sender As Object, ByVal e As EventArgs)
            Throw New UserFriendlyException("Action from custom view item was executed!")
        End Sub
    End Class
End Namespace
